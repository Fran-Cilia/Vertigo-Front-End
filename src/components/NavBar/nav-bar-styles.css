
.navBar{
    background: #000;
    height: 80px;
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: center;
    padding: 0.5rem calc((100vw - 1000px) / 2);
    z-index: 10;
}

.navLink {
    color: #fff;
    text-decoration: none;
    padding: 0.5rem 1rem;
    border-radius: .2pc;
}

.navLink:hover {
    background-color: #393a48;
    border-radius: 4px;
    transition: all 0.2s ease-out;
}

.logo{
    height: 120px;  
    padding: 0.5rem 1rem;
}






