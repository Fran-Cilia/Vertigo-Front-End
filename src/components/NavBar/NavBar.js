import React from "react";
import './nav-bar-styles.css';
import { Link, Route, Router, BrowserRouter } from 'react-router-dom'
import { FaBars } from 'react-icons/fa'

export const NavBar = () => {

    return (
        <div className="navBar">
            <img className="logo" data-testid="logo" src={require('../../Images/Logo.png')} alt="Logo" />
            <nav>
                <Link to="transactions" className="navLink" data-testid="payment">Pay or Request</Link>

                <Link to="transactions" className="navLink" data-testid="transaction">Transactions</Link>

                <Link to="profile" className="navLink" data-testid="profile">Profile</Link>

            </nav>

        </div>
    )
}

export default NavBar