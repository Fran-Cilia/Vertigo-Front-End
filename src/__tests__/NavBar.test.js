import { render, screen, cleanup } from '@testing-library/react'
import renderer, { act } from 'react-test-renderer'
import { MemoryRouter } from 'react-router';
import NavBar from '../components/NavBar/NavBar'

describe('NAVBAR RENDER TESTS', () => {

    afterEach(() => {
        cleanup();
    });

    it('Renders Navbar Components', () => {
        render(
            <MemoryRouter>
                <NavBar />
            </MemoryRouter>);

        //Logo
        const logo = screen.getByTestId('logo');
        expect(logo).toBeInTheDocument();

        //Pay or Request Link
        const payment = screen.getByTestId('payment');
        expect(payment).toBeInTheDocument();
        expect(payment).toHaveTextContent('Pay or Request')

        //Transactions Link
        const transaction = screen.getByTestId('transaction');
        expect(transaction).toBeInTheDocument();
        expect(transaction).toHaveTextContent('Transactions')

        //Profile Link
        const profile = screen.getByTestId('profile');
        expect(profile).toBeInTheDocument();
        expect(profile).toHaveTextContent('Profile')
    })

    it('Matches Snapshot', () => {
        const tree = renderer.create(
            <MemoryRouter>
                <NavBar />
            </MemoryRouter>).toJSON();
        expect(tree).toMatchSnapshot();
    })

})


// describe('NAVBAR FUNCTINOALITY TEST', () => {

//     it('Naviagates To Payments', () => {
//         const root = document.createElement('div');
//         document.body.appendChild(root);

//         render(
//             <MemoryRouter initialEntries={['/']}>
//                 <NavBar />
//             </MemoryRouter>,
//             root
//         );

//         act(() => {
//             const paymentsLink = document.querySelector('#paymentsLink');
//             paymentsLink.dispatchEvent(new MouseEvent("click"));
//         });

//         console.log(paymentsLink);
//         //expect(document.body.textContent).toBe('Welcome')
//     })
// })