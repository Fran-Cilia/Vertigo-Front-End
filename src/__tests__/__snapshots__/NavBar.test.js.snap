// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NAVBAR RENDER TESTS Matches Snapshot 1`] = `
<div
  className="navBar"
>
  <img
    alt="Logo"
    className="logo"
    data-testid="logo"
    src="Logo.png"
  />
  <nav>
    <a
      className="navLink"
      data-testid="payment"
      href="/transactions"
      onClick={[Function]}
    >
      Pay or Request
    </a>
    <a
      className="navLink"
      data-testid="transaction"
      href="/transactions"
      onClick={[Function]}
    >
      Transactions
    </a>
    <a
      className="navLink"
      data-testid="profile"
      href="/profile"
      onClick={[Function]}
    >
      Profile
    </a>
  </nav>
</div>
`;

exports[`NAVBAR TESTS Matches Snapshot 1`] = `
<div
  className="navBar"
>
  <img
    alt="Logo"
    className="logo"
    data-testid="logo"
    src="Logo.png"
  />
  <nav>
    <a
      className="navLink"
      data-testid="payment"
      href="/transactions"
      onClick={[Function]}
    >
      Pay or Request
    </a>
    <a
      className="navLink"
      data-testid="transaction"
      href="/transactions"
      onClick={[Function]}
    >
      Transactions
    </a>
    <a
      className="navLink"
      data-testid="profile"
      href="/profile"
      onClick={[Function]}
    >
      Profile
    </a>
  </nav>
</div>
`;
